src/bugsplat/bugsplat.py

```diff
@@ -1,100 +1,120 @@
 import json
-import os
-import requests
 import traceback
-import sys
+from logging import Logger, getLogger
+from os import PathLike
+from pathlib import PurePath
+from typing import List
+
+import requests
+
 
 class BugSplat:
-    def __init__(self, database, application, version):
+    def __init__(self,
+                 database: str,
+                 application: str,
+                 version: str,
+                 logger: Logger = None):
         self.database = database
         self.application = application
         self.version = version
-        self.additionalFilePaths = []
-        self.appKey = ''
+        self.additional_file_paths = []
+        self.app_key = ''
         self.description = ''
         self.email = ''
         self.user = ''
+        self.logger = logger or getLogger(__name__)
 
-    def setDefaultAdditionalFilePaths(self, additionalFilePaths):
-        self.additionalFilePaths = additionalFilePaths
+    def set_default_additional_file_paths(self, additional_file_paths: List[PathLike]):
+        self.additional_file_paths = additional_file_paths
 
-    def setDefaultAppKey(self, key):
-        self.appKey = key
+    def set_default_app_key(self, key: str):
+        self.app_key = key
 
-    def setDefaultDescription(self, description):
+    def set_default_description(self, description: str):
         self.description = description
-            
-    def setDefaultEmail(self, email):
+
+    def set_default_email(self, email: str):
         self.email = email
 
-    def setDefaultUser(self, user):
+    def set_default_user(self, user: str):
         self.user = user
 
-    def post(self, ex, additionalFilePaths = [], appKey = '', description = '', email = '', user = ''):
-        print('\n')
-        print('BugSplat caught an Unhandled Exception!')
-        print('\n')
+    # TODO Remove these completely
+    # left as aliases so nothing breaks
+    setDefaultAdditionalFilePaths = set_default_additional_file_paths
+    setDefaultAppKey = set_default_app_key
+    setDefaultDescription = set_default_description
+    setDefaultEmail = set_default_email
+    setDefaultUser = set_default_user
+
+    def post(self,
+             ex: BaseException or str,
+             additional_file_paths: List[PathLike] = None,
+             app_key: str = '',
+             description: str = '',
+             email: str = '',
+             user: str = ''):
+
+        if not additional_file_paths:
+            additional_file_paths = self.additional_file_paths
+
+        self.logger.warning('\nBugSplat caught an Unhandled Exception!\n')
 
         # TODO BG what if ex is not defined? Do we care?
-        # https://stackoverflow.com/questions/3702675/how-to-catch-and-print-the-full-exception-traceback-without-halting-exiting-the
-        callstack = self._convertExceptionToJson(ex)
-        exceptionMessage = str(ex)
+        # https://stackoverflow.com/questions/3702675/how-to-catch-and-print-the-full
```

Comment:
I think the linter broke up a URL, can we put this back on a single line?

Reply:
Oops, good catch! What about if I update it with the perma link which is shorter?